build-caching/src/main/java/com/vertispan/j2cl/build/TaskScheduler.java

```diff
@@ -211,7 +223,27 @@ private void executeTask(CollectedTaskInputs taskDetails, DiskCache.CacheResult
                     }
                     try {
                         long start = System.currentTimeMillis();
-                        taskDetails.getTask().execute(new TaskContext(result.outputDir(), log));
+
+                        Optional<DiskCache.CacheResult> latestResult = buildCache.getLatestResult(taskDetails.getProject(), taskDetails.getTaskFactory().getOutputType());
+                        final TaskSummaryDiskFormat taskSummaryDiskFormat = latestResult.map(TaskScheduler.this::getTaskSummary).orElse(null);
+                        // deserialize its inputs list and use its outputs below
+                        for (Input input : taskDetails.getInputs()) {
```

Comment:
Need to return to this and invalidate the cache entry if it turns out there used to be (in `taskSummaryDiskFormat`) a dependency that is no longer present in `taskDetails.getInputs()`, else we won't notice the individual changes that took place in that jar (that is, being entirely removed).

Reply:
This is a blocker to merging this.

Reply:
This was resolved in 3c9da784df5746ce65493f84b86a03e83cc0ebe1.